pyquil/tests/test_quilt.py

```diff
@@ -195,6 +197,68 @@ def test_program_calibrate():
     assert calibrated == Program('SHIFT-PHASE 0 "rf" -pi').instructions
 
 
+def test_program_calibrate_recursive():
+    # Not a realistic example :)
+    prog = Program(
+        """
+DEFCAL RX(%theta) q:
+    RY(%theta) q
+
+DEFCAL RZ(%theta) q:
+    RX(%theta) q
+"""
+    )
+    calibrated = prog.calibrate(Gate("RZ", [np.pi], [Qubit(0)]))
+    assert calibrated == Program("RY(pi) 0").instructions
+
+
+def test_program_calibrate_cyclic_error():
```

Comment:
would prefer separate or parametrized test cases for these independent programs, i.e.

```python
@pytest.mark.parametrize('program_text', ('DEFCAL RZ(%theta) q:\n\tRZ(%theta) q', ...))
def test_program_calibrate_cyclic_error():
    ...
```